content/developer/vanilla-cli/build-process.md

```diff
@@ -0,0 +1,109 @@
+
+#### Javascript
+
+The single javascript entry will be bundled and outputted to `js/custom.js`. See [Processing Javascript] for more information on how the build process works.
+
+#### Images
+
+Images are outputted with the same filenames and structure as their entries into the `design/images` folder.
+
+### Example Folder Structure
+
+This build process relies on the addon having the following structure for its source files:
+
+```
+addonfoldername
+|--src
+   |--js
+      |--index.js
+      |--another-file.js
+   |--scss
+      |--some-entry-file.scss
+      |--_not-an-entry-file.scss
+   |--images
+      |--image1.png
+      |--image2.jpg
+```
+
+Running a build on the given folder structure would would result in the following output files
+
+```
+addonfoldername
+|--design
+   |--some-entry-file.css
+   |--images
+      |--image1.png
+      |--image2.jpg
+|--design
+   |--custom.js
+```
+
+### Bundling Process
+
+See the [Bundling Process](/developer/vanilla-cli/bundling-process) page for details about the CLI bundles it's assets.
+
+## Legacy
+
+### Notice
+
+This build process is deprecated and only exists to support legacy addons. New projects should use the latest numbered build process. 
+
+- It requires whatever tooling the addon requires. This could be node, yarn/npm, or ruby.
+- It may require other global node dependancies such as `gulp`, `grunt`, or `bower`.
+- It may require installation of ruby gems or node modules before building.
+
+**This build process is not a magic bullet. It will not fix a broken legacy build process. It is only a wrapper around existing ones.**
+
+This build process will attempt to find the tasks `build` and `watch` tasks by:
+1. Checking if any `npm scripts` exists named `build` or `watch` and running those.
+2. Then checking and running any of the following if they exist. You can have multiple of these
+- `gulp default`, `gulp build`, or `gulp watch` tasks.
+- `grunt default`, `grunt build`, or `grunt watch` tasks
+- If a Gemfile exists, try to build ruby-sass with [Compass](https://github.com/Compass/compass) using `compass compile` or `compass watch`
+
+This legacy build process will also create an empty `bower_components` if a `bower.json` file exists and the folder does not. This is to prevent certain build process from crashing even if you have no bower components.
+
+This process will **not** install your `node_modules` or `bower_components` for you. You are responsible for those yourself and they can normally be handled by running `yarn install`.
+
+If there are major issues with a legacy build process consider converting it to use one of the [built in processes](/developer/vanilla-cli/build-quickstart).
```

Comment:
I would add a page or a section here about legacy builds. Just list them off and how to run them (compass, gulp, grunt, etc). Mark them all as deprecated. It's good to have this info just in case Vanilla Cli fails and you need to run it the old way. Maybe a quick guide to how to figure out the right way to run them. A note about npm vs yarn would also be good.

Reply:
A blurb about less would also be a good idea, seeing as some of our themes use it.

Reply:
I feel deeper instructions about legacy processes should be left to internal docs and training. The purpose of this tool isn't to teach someone how to use npm or ruby, it's specifically to share a build process.

I've updated the legacy docs to try and emphasize that it is the least significant part of this tool and is mostly a helper. TBH in a lot of cases the issues are with broken dependencies in the legacy process (a lot our grunt processes are using 5-10 packages with versions <= 1.0) and migrating to use the built in process can be easier than trying to patch together the a broken build. I've put some significant effort into the Quickstart/conversion guide as a result.

The emphasis should be on ripping out that tech-debt instead of wrapping bandages around it.